nms/app/packages/magmalte/app/views/subscriber/SubscriberChart.js

```diff
@@ -0,0 +1,330 @@
+  const {start, end, networkId, subscriberId} = props;
+  const [delta, unit, format] = getStep(start, end);
+  let requestError = '';
+  const step = getStepString(delta, unit);
+  const toolTipHint = delta + ' ' + unit;
+
+  const queries = [
+    {
+      q: `sum(increase(ue_traffic{IMSI="${subscriberId}",direction="down"}[${step}]))`,
+      color: colors.secondary.dodgerBlue,
+      label: 'download',
+    },
+    {
+      q: `sum(increase(ue_traffic{IMSI="${subscriberId}",direction="up"}[${step}]))`,
+      color: colors.data.flamePea,
+      label: 'upload',
+    },
+  ];
+  const allDatasets = [];
+  const labels = [];
+  let addLabels = true;
+  const requests = queries.map(async (query, index) => {
+    try {
+      const resp = await MagmaV1API.getNetworksByNetworkIdPrometheusQueryRange({
+        networkId: networkId,
+        start: start.toISOString(),
+        end: end.toISOString(),
+        step: getStepString(delta, unit),
+        query: query.q,
+      });
+
+      const data = [];
+      resp.data.result.forEach(it =>
+        it['values']?.map(i => {
+          if (addLabels) {
+            labels.push(moment.unix(parseInt(i[0])).format(format));
+          }
+          data.push({
+            t: parseInt(i[0]) * 1000,
+            y: parseFloat(i[1]),
+          });
+        }),
+      );
+      addLabels = false;
+      allDatasets.push({
+        datasetKeyProvider: index.toString(),
+        label: query.label,
+        fill: false,
+        barPercentage: 0.7,
+        pointHitRadius: 10,
+        pointRadius: 2,
+        borderWidth: 2,
+        backgroundColor: query.color,
+        borderColor: query.color,
+        hoverBackgroundColor: query.color,
+        hoverBorderColor: 'black',
+        data: data,
+        unit: 'bytes',
+      });
+    } catch (error) {
+      requestError = error;
+      return [];
+    }
+  });
+
+  await Promise.all(requests);
+  if (requestError) {
+    props.enqueueSnackbar('Error getting event counts', {
+      variant: 'error',
+    });
+  }
+  return {allDatasets, labels, unit, toolTipHint};
+}
+
+function SubscriberDataKPI() {
+  const {match} = useRouter();
+  const networkId: string = nullthrows(match.params.networkId);
+  const subscriberId: string = nullthrows(match.params.subscriberId);
+  const [usage1H, setUsage1H] = useState([0, 'bytes']);
+  const [usage1D, setUsage1D] = useState([0, 'bytes']);
+  const [usage1W, setUsage1W] = useState([0, 'bytes']);
+  const [usage1M, setUsage1M] = useState([0, 'bytes']);
+  const [isLoading, setIsLoading] = useState(true);
+  const enqueueSnackbar = useEnqueueSnackbar();
+
+  useEffect(() => {
+    // fetch queries
+    const fetchAllData = async () => {
+      const [
+        dataUsage1h,
+        dataUsage1d,
+        dataUsage1w,
+        dataUsage1m,
+      ] = await Promise.allSettled([
+        MagmaV1API.getNetworksByNetworkIdPrometheusQuery({
+          networkId: networkId,
```

Comment:
```suggestion
          networkId,
```
this works in JS I think?